command/app.go

```diff
@@ -77,9 +77,9 @@ var app = &cli.App{
 			NoVerifySSL: noVerifySSL,
 		}
 
-		storage.SetS3Options(s3opts)
+		err := storage.SetS3Options(s3opts)
```

Comment:
If we cannot create a session at the beginning, whole program will fail. In previous version, each job was trying to reuse the first successful session and create a new session if there is an error on previous attempts. I think this approach is so much cleaner since we don't need to handle errors in everywhere. But I am not sure we can safely exit the program here. What do you think?

Reply:
I think; using a  `fail fast`  approach is not a bad choice, since creating an s3 session is at the core of this tool.